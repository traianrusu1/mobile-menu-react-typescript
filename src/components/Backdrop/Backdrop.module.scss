@import '../MobileMenu/mobileMenuVariables';

.backdrop {
  width: 100vw;
  height: 100vh;
  position: absolute;
  z-index: 100;
  left: 0;
  top: $toolbarHeight;
  background-color: rgba(100, 100, 100, 0.5);
}
