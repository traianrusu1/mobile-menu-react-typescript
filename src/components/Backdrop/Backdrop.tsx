/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import styles from './Backdrop.module.scss';

interface Props {
  show: boolean;
  onBackDropClick: () => void;
}

const Backdrop: React.FC<Props> = ({ show, onBackDropClick }: Props) => {
  return show ? <div className={styles.backdrop} onClick={onBackDropClick} /> : <></>;
};

export default Backdrop;
