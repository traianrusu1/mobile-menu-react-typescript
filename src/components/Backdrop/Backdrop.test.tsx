import React from 'react';
import { render, cleanup } from '@testing-library/react';

import Backdrop from './Backdrop';

afterEach(cleanup);

describe('<Backdrop />', () => {
  test('renders', () => {
    const { container } = render(<Backdrop myProp="" />);
    expect(container).toMatchSnapshot();
  });
});

