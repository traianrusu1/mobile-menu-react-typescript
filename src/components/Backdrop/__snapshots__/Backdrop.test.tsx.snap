// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop /> renders 1`] = `
<div>
  <div
    class="backdrop"
  >
    Backdrop
    
  </div>
</div>
`;
